src/store/undoRedo/deeplink/index.ts

```diff
@@ -38,8 +38,19 @@ export function getDeepLink({changedKeys, flows}: {changedKeys: string[], flows:
     }
   } else if (BLOCK_CHANGE_REGEX.test(key)) {
     const {flowIndex, blockIndex, fieldPath} = parseBlockChangeParams(key)!
-    // TODO in CORE-664: fix block deletion which triggers a console error here because flows.flows[flowIndex].blocks[blockIndex] is undefined
-    const blockId = flows.flows[flowIndex].blocks[blockIndex].uuid
+    const blockId = flows.flows[flowIndex].blocks[blockIndex]?.uuid ?? null
+
+    if (blockId === null) {
```

Comment:
Nice !

- [ ]  Could you pls leave a comment we're having this scenario for block deletion for eg.?